src/GraphNavigator/SerializationGraphNavigator.php

```diff
@@ -234,6 +234,10 @@ public function accept($data, ?array $type = null)
                     throw new ExcludedClassException();
                 }
 
+                if (!is_object($data)) {
+                    throw new InvalidArgumentException('Value at '.$this->context->getPath().' is expected to be an object of class '.type['name'].' but is of type '.gettype($data));
+                }
```

Comment:
the startVisitingObject below has the type declaration `object $data`, so we get a TypeError exception if we call it with a non-object.

Reply:
would probably need some setup, especially for a unit test. i think annotating a model property to be some object and assigning an array to it (or a string/int/bool) and then running serialization should trigger it.